pkg/virt-controller/services/template.go

```diff
@@ -251,6 +251,33 @@ func (t *templateService) RenderLaunchManifest(vmi *v1.VirtualMachineInstance) (
 		})
 	}
 
+	// Handle CPU pinning
+	cpus := vmi.Spec.Domain.CPU
+	if cpus != nil && cpus.DedicatedCPUPlacement {
+		// schedule only on nodes with a running cpu manager
+		if vmi.Spec.NodeSelector == nil {
+			vmi.Spec.NodeSelector = make(map[string]string)
+		}
+		vmi.Spec.NodeSelector[v1.CPUManager] = "true"
+
+		if resources.Limits == nil {
+			resources.Limits = make(k8sv1.ResourceList)
+		}
+		if cpus.Cores != 0 {
+			resources.Limits[k8sv1.ResourceCPU] = *resource.NewQuantity(int64(cpus.Cores), resource.BinarySI)
+			resources.Requests[k8sv1.ResourceCPU] = *resource.NewQuantity(int64(cpus.Cores), resource.BinarySI)
+		} else {
+			if cpuLimit, ok := resources.Limits[k8sv1.ResourceCPU]; ok {
```

Comment:
I just wonder if we need to check that `resources` can be converted to an integer because a user can to specify
- `cpus.DedicatedCPUPlacement`
- `resources.Limits` equal to something like 200m

In this case I think you will fail in converter, so better to add this check to validation webhook

Reply:
Absolutely. I've added several validations but didn't post yet as I'm still struggling with the functional tests.

Also, I wanted to explicitly check, somewhere in the SyncVM that the pods QOS is `Guaranteed`, but didn't figure out yet how to get to pods informer at that stage.